checkov/common/util/type_forcers.py

```diff
@@ -44,15 +45,16 @@ def force_float(var: Any) -> float | None:
         return None
 
 
-def convert_str_to_bool(bool_str: bool | str) -> bool | str:
+def convert_str_to_bool(bool_str: bool | str) -> bool:
     if isinstance(bool_str, str):
         bool_str_lower = bool_str.lower()
         if bool_str_lower in ("true", '"true"'):
             return True
         elif bool_str_lower in ("false", '"false"'):
             return False
 
-    return bool_str
+    # If we got here it must be a boolean, mypy doesn't understand it, so we use cast
```

Comment:
just use else and you won't need it

Reply:
Tried it, doesn't work as it waits for a return value as well outside of the else

Reply:
Fixed